src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativeCustomWorkingDir.groovy

```diff
@@ -0,0 +1,38 @@
+pipeline {
+    agent {
+        kubernetes {
+            defaultContainer 'maven'
+            yaml """
+metadata:
+  labels:
+    some-label: some-label-value
+    class: KubernetesDeclarativeAgentTest
+spec:
+  containers:
+  - name: jnlp
+    env:
+    - name: CONTAINER_ENV_VAR
+      value: jnlp      
+  - name: maven
+    image: maven:3.3.9-jdk-8-alpine
+    workingDir: /home/jenkins/wsp1
+    command:
+    - cat
+    tty: true
+    env:
+    - name: CONTAINER_ENV_VAR
+      value: maven
+"""
+        }
+    }
+
+    stages {
+        stage('Run maven') {
+            steps {
+                sh 'mvn -version'
+                sh "echo Workspace dir is ${pwd()}"
+                sh "echo \$WORKSPACE"
+            }
+        }
+    }
+}
```

Comment:
I suggest the following changes to cover more ground. The java part should assert with working dirs.

```
diff --git a/src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativeCustomWorkingDir.groovy b/src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativeCustomWorkingDir.groovy
index 72034c55..d4a84771 100644
--- a/src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativeCustomWorkingDir.groovy
+++ b/src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativeCustomWorkingDir.groovy
@@ -1,7 +1,6 @@
 pipeline {
     agent {
         kubernetes {
-            defaultContainer 'maven'
             yaml """
 metadata:
   labels:
@@ -29,9 +28,15 @@ spec:
     stages {
         stage('Run maven') {
             steps {
-                sh 'mvn -version'
-                sh "echo Workspace dir is ${pwd()}"
-                sh "echo \$WORKSPACE"
+                dir('foo') {
+                    sh 'echo [jnlp] current dir is $(pwd)'
+                    sh 'echo [jnlp] WORKSPACE=$WORKSPACE'
+                    container('maven') {
+                        sh 'mvn -version'
+                        sh 'echo [maven] current dir is $(pwd)'
+                        sh 'echo [maven] WORKSPACE=$WORKSPACE'
+                    }
+                }
             }
         }
     }

```

Reply:
Thanks for the suggestion. I took the example groovy from the JENKINS bug. I'll enhance the test case.